### PR TITLE
Fix tag parsing

### PR DESCRIPTION
**Pull request by gutjuri**
I already opened an issue and a PR in the original repo of this project (see https://github.com/alecthomas/jsonschema/issues/117). 

Since this seems to be the currently maintained fork of the project, I wanted to share this bugfix with you. I'd be glad if you accepted it.

I'm open for criticism and comments.

### CODE REVIEW

File: reflect.go [hunk 1]
```diff
@@ -890,6 +891,23 @@ func (r *Reflector) typeName(t reflect.Type) string {
...
 
+// Only split if the rune after the comma is a letter
+// This way, we prevent splitting regexes
+func splitOnComma(tagString string) []string {
+	tags := make([]string, 0)
+	var lastRune rune = 0
+	lastSplit := 0
+
+	for i, chr := range tagString {
+		if lastRune == ',' && unicode.IsLetter(chr) {
```
Comment (samlown) [inline reflect.go:902]: I have doubts about this approach. There is no guarantee that the next character after a comma is the separator. Have you tried using an escape character? i.e. `\`. You could then split on an regexp instead, with a pattern like `[^\\],`.

**Review by samlown — CHANGES_REQUESTED** [review]
This for sure is useful! In addition to my comment, I'd suggest adding some direct unit tests for the method to test out several scenarios. I don't think you need a full schema example for this one.

Also, we're currently working on a major refactor to move to JSON Schema Draft 2020-12. It'd be great if you could update your PR afterwards! Should be a simple merge for this use-case. Thank you!!

**Comment by gutjuri** [comment]
I updated my PR but it won't show up in this thread, because GH no longer considers this repo a fork of alecthomas/jsonschema. I'll refork and submit a new PR.

**Comment by samlown** [comment]
> I updated my PR but it won't show up in this thread, because GH no longer considers this repo a fork of alecthomas/jsonschema. I'll refork and submit a new PR.

You should able to just re-assign the PR to a new branch in the new repo, but please do what you think is easiest!

**Comment by gutjuri** [comment]
see #8 now

**Pull request closed without merging**